src/content/docs/features/dialog.mdx

```diff
@@ -5,10 +5,315 @@ description: Native system dialogs for opening and saving files along with messa
 
 import Stub from '@components/Stub.astro';
 import PluginLinks from '@components/PluginLinks.astro';
+import { Tabs, TabItem } from '@astrojs/starlight/components';
+import CommandTabs from '@components/CommandTabs.astro';
 
 <PluginLinks plugin="dialog" />
 
-<Stub>
-	Based on
-	https://github.com/tauri-apps/plugins-workspace/tree/v2/plugins/dialog
-</Stub>
+Native system dialogs for opening and saving files along with message dialogs.
+
+## Setup
+
+Install the dialog plugin to get started.
+
+<Tabs>
+    <TabItem label="Automatic">
+
+    1. Use your project's package manager to add the dependency:
+
+    <CommandTabs npm="npm run tauri add dialog"
+    yarn="yarn run tauri add dialog"
+    pnpm="pnpm tauri add dialog"
+    cargo="cargo tauri add dialog" />
+
+    2. Modify `lib.rs` to initialize the plugin:
+
+    ```rust
+    #[cfg_attr(mobile, tauri::mobile_entry_point)]
+    pub fn run() {
+        tauri::Builder::default()
+            // Initialize the plugin
+            .plugin(tauri_plugin_dialog::init())
+            .run(tauri::generate_context!())
+            .expect("error while running tauri application");
+    }
+    ```
+    </TabItem>
+    <TabItem label="Manual">
+
+    1. Run `cargo add tauri-plugin-dialog` to add the plugin to the project's dependencies in `Cargo.toml`.
+
+    2. Modify `lib.rs` to initialize the plugin:
+
+    ```rust
+    // lib.rs
+    #[cfg_attr(mobile, tauri::mobile_entry_point)]
+    pub fn run() {
+        tauri::Builder::default()
+            // Initialize the plugin
+            .plugin(tauri_plugin_dialog::init())
+            .run(tauri::generate_context!())
+            .expect("error while running tauri application");
+    }
+    ```
+
+    3. If you'd like create dialogs in JavaScript, install the npm package as well:
+
+    <CommandTabs
+        npm="npm install @tauri-apps/plugin-dialog"
+        yarn="yarn add @tauri-apps/plugin-dialog"
+        pnpm="pnpm add @tauri-apps/plugin-dialog"
+    />
+
+</TabItem>
+</Tabs>
+
+## Usage
+
+Here's how you can use the dialog plugin:
+
+- [Create Yes/No Dialog](#create-yesno-dialog)
+- [Create Ok/Cancel Dialog](#create-okcancel-dialog)
+- [Create Message Dialog](#create-message-dialog)
+- [Open a File Selector Dialog](#open-a-file-selector-dialog)
+- [Save to File Dialog](#save-to-file-dialog)
+
+{/* TODO: Link to which language to use, frontend vs. backend guide when it's made */}
+
+The dialog plugin is available in both JavaScript and Rust.
+
+{/* ASK */}
+
+### Create Yes/No Dialog
+
+Shows a question dialog with `Yes` and `No` buttons
+
+<Tabs>
+<TabItem label="JavaScript">
+
+```js
+import { ask } from '@tauri-apps/plugin-dialog';
+
+// Create a Yes/No dialog
+const yes = await ask('This action cannot be reverted. Are you sure?', {
```

Comment:
I would rename this to something like `answer` since it could be either true or false and then update what's logged below.